### ISSUE TITLE
About the released checkpoints

### ISSUE THREAD

**Issue by ZhihongShao** [allanj/Deductive-MWP#1]
It seems that the released checkpoints for MathQA and SVAMP have significantly worse performance than reported in the paper.

**Comment by allanj** [comment]
may I know how  you run the model? I'm happy to help

**Comment by allanj** [comment]
I also just quickly run the experiment on MathQA even using CPU for inference.
![image](https://user-images.githubusercontent.com/3351187/163686073-079a57d6-0385-477c-929b-bee602581be2.png)

I still get the same value accuracy.

Maybe one thing I need to change is that the argument here is `dev_file`. but I should change to `test_file`
https://github.com/allanj/Deductive-MWP/blob/main/universal_main_ddp.py#L464

But I'm not sure if this is the problem in your case.


Feel free to ask for help

For svamp, I supposed you run the version that without constraint that would be around 45.

**Comment by ZhihongShao** [comment]
I have tried the following two commands, and got the same results which are worse than reported.

```bash
use_constant=1
add_replacement=1
consider_multiple_m0=1
var_update_modes=(gru)
cuda_devices=(0)
model_folder=mathqa
bert_folder=/home/shaozhihong/.cache/torch
bert_model_name=roberta-base

python3 universal_main.py --device=cuda:6 \
                    --model_folder=${model_folder} \
                    --mode=test \
                    --height=10 \
                    --train_max_height=15 \
                    --num_epochs=1000 \
                    --consider_multiple_m0=${consider_multiple_m0} \
                    --train_file=data/MathQA/mathqa_train_nodup_our_filtered.json \
                    --dev_file=data/MathQA/mathqa_test_nodup_our_filtered.json \
                    --batch_size=30 \
                    --add_replacement=${add_replacement} \
                    --train_num=-1 \
                    --dev_num=-1  \
                    --var_update_mode=${var_update_mode} \
                    --bert_folder=${bert_folder} \
                    --bert_model_name=${bert_model_name} \
                    --use_constant=${use_constant} \
                    --fp16=1  \
                    --parallel=1 \
                    --learning_rate=2e-5 > logs/${model_folder}_test.log 2>&1 &

accelerate launch --mixed_precision=fp16 universal_main_ddp.py --device=cuda:3 \
                    --model_folder=${model_folder} \
                    --mode=test \
                    --height=15 \
                    --train_max_height=15 \
                    --num_epochs=1000 \
                    --consider_multiple_m0=${consider_multiple_m0} \
                    --train_file=data/MathQA/mathqa_train_nodup_our_filtered.json \
                    --dev_file=data/MathQA/mathqa_test_nodup_our_filtered.json \
                    --batch_size=30 \
                    --add_replacement=${add_replacement} \
                    --train_num=-1 \
                    --dev_num=-1  \
                    --var_update_mode=${var_update_mode} \
                    --bert_folder=${bert_folder} \
                    --bert_model_name=${bert_model_name} \
                    --use_constant=${use_constant} \
                    --fp16=1  \
                    --parallel=1 \
                    --learning_rate=2e-5 > logs/${model_folder}_ddp_test.log 2>&1 &
```

```bash
04/18/2022 14:58:57 - INFO - __main__ -   before removed: 1605, after removed: 1605
04/18/2022 14:58:57 - INFO - __main__ -   [Info] Equation accuracy: 5.17%, total: 1605, corr: 83, adjusted_total: 1605
04/18/2022 14:58:57 - INFO - __main__ -   [Info] Value accuracy: 6.23%, total: 1605, corr: 100, adjusted_total: 1605
04/18/2022 14:58:57 - INFO - __main__ -   [Info] step num: 5 Acc.:0.00 (0/198) val acc: 1.52 (3/198)
04/18/2022 14:58:57 - INFO - __main__ -   [Info] step num: 3 Acc.:0.00 (0/391) val acc: 1.79 (7/391)
04/18/2022 14:58:57 - INFO - __main__ -   [Info] step num: 6 Acc.:0.00 (0/156) val acc: 1.28 (2/156)
04/18/2022 14:58:57 - INFO - __main__ -   [Info] step num: 2 Acc.:0.00 (0/237) val acc: 0.00 (0/237)
04/18/2022 14:58:57 - INFO - __main__ -   [Info] step num: 4 Acc.:0.00 (0/289) val acc: 0.35 (1/289)
04/18/2022 14:58:57 - INFO - __main__ -   [Info] step num: 7 Acc.:0.00 (0/87) val acc: 1.15 (1/87)
04/18/2022 14:58:57 - INFO - __main__ -   [Info] step num: 8 Acc.:0.00 (0/57) val acc: 1.75 (1/57)
04/18/2022 14:58:57 - INFO - __main__ -   [Info] step num: 13 Acc.:0.00 (0/5) val acc: 0.00 (0/5)
04/18/2022 14:58:57 - INFO - __main__ -   [Info] step num: 9 Acc.:0.00 (0/36) val acc: 0.00 (0/36)
04/18/2022 14:58:57 - INFO - __main__ -   [Info] step num: 1 Acc.:75.45 (83/110) val acc: 76.36 (84/110)
04/18/2022 14:58:57 - INFO - __main__ -   [Info] step num: 12 Acc.:0.00 (0/14) val acc: 0.00 (0/14)
04/18/2022 14:58:57 - INFO - __main__ -   [Info] step num: 11 Acc.:0.00 (0/12) val acc: 8.33 (1/12)
04/18/2022 14:58:57 - INFO - __main__ -   [Info] step num: 10 Acc.:0.00 (0/11) val acc: 0.00 (0/11)
04/18/2022 14:58:57 - INFO - __main__ -   [Info] step num: 15 Acc.:0.00 (0/1) val acc: 0.00 (0/1)
04/18/2022 14:58:57 - INFO - __main__ -   [Info] step num: 14 Acc.:0.00 (0/1) val acc: 0.00 (0/1)
```

**Comment by allanj** [comment]
I just tried your second command, but also need to add one more line before evaluation
```python
model, valid_dataloader = accelerator.prepare(model, valid_dataloader)
```
But I believe this is not your problem because it will raise errors while running.

I can't really reproduce your results, but there should be some errors in your case. I will try to create a colab notebook for you

**Comment by allanj** [comment]
https://colab.research.google.com/drive/1O_3byrRA93PrHK6s1bVdZePdr5m4HOXL?usp=sharing 

You can refer to this simple notebook for more information.

**Comment by allanj** [comment]
I tried changing the batch size from 8 to 30, got consistent results.

![image](https://user-images.githubusercontent.com/3351187/163785404-5df4bb42-192b-47c0-a402-67676e6b845a.png)

**Comment by ZhihongShao** [comment]
That's strange. I didn't change anything. The commands I used are from the scripts you provided. Is it possible that this github repository or the uploaded checkpoints are different from your local files?

Thanks for providing the notebook! It would be helpful to find out what went wrong. I have applied for access to it :)

**Comment by allanj** [comment]
I also used the checkpoint in this repo, so it should be fine. The address of the checkpoint in the notebook is also the same as the one in this repo.

You can try to run the notebook first, and see how I can help

**Comment by allanj** [comment]
Feel free to open the issue again if you still run into errors in this

**Comment by ZhihongShao** [comment]
I can reproduce your results now. Thanks!
